src/drivers/sqs/Queue.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace yii\queue\sqs;
+
+use yii\base\NotSupportedException;
+use yii\queue\cli\LoopInterface;
+use yii\queue\cli\Queue as CliQueue;
+use yii\queue\cli\Signal;
+use \Aws\Sqs\SqsClient;
+use Aws\Credentials\CredentialProvider;
+
+/**
+ * SQS Queue
+ *
+ * @author Manoj Malviya <manojm@girnarsoft.com>
+ */
+class Queue extends CliQueue
+{
+    /**
+    * @var SqsClient
+    */
+    private $_client;
+
+    /**
+     * The SQS url.
+     * @var string
+     */
+    public $url;
+
+    /**
+     * aws access key
+     * @var string
+     */
+    public $key = '';
+
+    /**
+     * aws secret
+     * @var string
+     */
+    public $secret = '';
+
+    /**
+     * region where queue is hosted.
+     * @var string
+     */
+    public $region = '';
+
+    /**
+     * API version
+     * @var string
+     */
+    public $version = 'latest';
+
+    /**
+     * @var string command class name
+     */
+    public $commandClass = Command::class;
+
+    /**
+     * @inheritdoc
+     */
+     public function init()
+     {
+         parent::init();
+     }
+
+    /**
+     * Listens queue and runs each job.
+     *
+     * @param bool $repeat whether to continue listening when queue is empty.
+     * @param int $delay number of seconds to sleep before next iteration.
+     * @return null|int exit code.
+     * @internal for worker command only
+     */
+    public function run($repeat, $delay = 0)
+    {
+        return $this->runWorker(function (LoopInterface $loop) use ($repeat, $delay) {
+            while ($loop->canContinue()) {
+                if ($payload = $this->getPayload()) {
+                    list($ttr, $message) = explode(';', $payload['Body'], 2);
+                    //reserve it so it is not visible to another worker till ttr
+                    $this->reserve($payload, $ttr);
+
+                    if ($this->handleMessage(null, $message, $ttr, 1)) {
+                        $this->release($payload);
+                    }
+                } elseif (!$repeat) {
+                    break;
+                } elseif ($delay) {
+                    sleep($delay);
+                }
+            }
+        });
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function pushMessage($message, $ttr, $delay, $priority)
+    {
+        if ($priority) {
+            throw new NotSupportedException('Priority is not supported in this driver');
+        }
+
+        $model = $this->getClient()->sendMessage([
+            'DelaySeconds' => $delay,
+            'QueueUrl' => $this->url,
+            'MessageBody' => "$ttr;$message",
+        ]);
+
+        if ($model === null) {
```

Comment:
In which cases `this->getClient()->sendMessage()` can return `null` instead of array?
If this is possible, `pushMessage` must throw an exception, because it always returns message ID. If no, the block should be removed.

Reply:
Correct, due to the documentation it returns model only. Exceptions are raised in aws sdk methods. I removed If block.